stories/molecules/SelectBadge.stories.tsx

```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import { Option, Priority, SelectBadge } from '@textkernel/oneui';
+
+const meta: Meta<typeof SelectBadge> = {
+    title: 'Molecules/SelectBadge',
+    component: SelectBadge,
+};
+
+export default meta;
+
+type Story = StoryObj<typeof SelectBadge>;
+
+const commonArgs = {
+    children: 'London',
+    option: { value: '15', label: '+15 km' },
+    optionList: [
+        { value: '5', label: '+5 km' },
+        { value: '15', label: '+15 km' },
+        { value: '25', label: '+25 km' },
+    ],
+    optionListHeader: 'Radius',
+};
+
+export const _SelectBadge: Story = {
+    name: 'SelectBadge',
+    args: {
+        ...commonArgs,
+        priority: 'mandatory',
+        priorityLabels: {
+            mandatory: 'Mandatory',
+            important: 'Important',
+            optional: 'Optional',
+            exclude: 'Exclude',
+        },
+    },
+    render: (args) => {
+        const [priority, setPriority] = React.useState<Priority>(args.priority);
+        const [selectedOption, setSelectedOption] = React.useState(args.option);
+
+        React.useEffect(() => {
+            setPriority(args.priority);
+        }, [args.priority]);
+
+        React.useEffect(() => {
+            setSelectedOption(args.option);
+        }, [args.option]);
+
+        const handlePriorityChange = (newPriority: Priority) => {
+            setPriority(newPriority);
+        };
+
+        const handleOptionChange = (newOption: Option) => {
+            setSelectedOption(newOption);
+        };
+
+        const handleClose = () => {
+            console.log('SelectBadge closed');
+        };
+
+        return (
```

Comment:
I don't think this example is realistic. If this was a real implementation, you would want to be able to change both `London` and the radius separately. hence the content of the main dropdown can be quite varied, and should not be handled by this component.